rest-api-spec/src/main/resources/rest-api-spec/test/search/360_number_field_term_doc_values_only.yml

```diff
@@ -0,0 +1,167 @@
+setup:
+    - skip:
+        features: [ "headers" ]
+        version: " - 2.11.99"
+        reason: "searching with only doc_values was added in 2.12.0"
+---
+"search on number fields with doc_values enabled":
```

Comment:
This test case only covers the exact match cases. Can we please cover the range and set cases too?

Since there are a lot of types here (including exotic ones like `half_float` and `unsigned_long`), each with their own implementation changed by this PR, I feel like we should at least check the happy path for every query type against every field type.

Reply:
@msfroh I've merged different doc_values tests into a single yaml test that runs against all three different combination of index and doc_values. I've also added the happy paths for all the numeric fields.

Reply:
Nice, thanks!